pkg/util/util.go

```diff
@@ -1041,17 +1041,68 @@ func DownloadFile(params DownloadParams) error {
 	return nil
 }
 
+// Load a file into memory (http(s):// or file://)
+func LoadFileIntoMemory(URL string) (fileBytes []byte, err error) {
+	// check if we have a file url
+	if strings.HasPrefix(strings.ToLower(URL), "file://") {
+		// strip off the "file://" to get a local filepath
+		filepath := strings.Replace(URL, "file://", "", -1)
+
+		// if filepath doesn't start with a "/"" then we have a relative
+		// filepath and will need to prepend the current working directory
+		if !strings.HasPrefix(filepath, "/") {
+			// get the current working directory
+			cwd, err := os.Getwd()
+			if err != nil {
+				return nil, errors.New("unable to determine current working directory")
+			}
+			// prepend the current working directory to the relatove filepath
+			filepath = fmt.Sprintf("%s/%s", cwd, filepath)
+		}
+
+		// check to see if filepath exists
+		info, err := os.Stat(filepath)
+		if os.IsNotExist(err) || info.IsDir() {
+			return nil, errors.New(fmt.Sprintf("unable to read file: %s, %s", URL, err))
+		}
+
+		// read the bytes from the filepath
+		fileBytes, err = ioutil.ReadFile(filepath)
+		if err != nil {
+			return nil, errors.New(fmt.Sprintf("unable to read file: %s, %s", URL, err))
+		}
+
+		return fileBytes, nil
+	} else {
+		// assume we have an http:// or https:// url and validate it
+		err = ValidateURL(URL)
```

Comment:
Could we make this work also with local paths?
In some cases, I would like to have a deploy manifest as local file in my repository. 
For example `file://./deploy.yaml`

Reply:
Do you mean the `ValidateURL` function? We can load a local file as specified here: https://github.com/openshift/odo/pull/3478/files/0c88e9d787b28a9d8dd50f3126671de497b35648#diff-6c6a7fefe37261e2fa8a010cf9e7ae08R1045-R1049